cirq-core/cirq/ops/control_values.py

```diff
@@ -180,3 +180,80 @@ def __and__(self, other: AbstractControlValues) -> 'ProductOfSums':
                 f'And operation not supported between types ProductOfSums and {type(other)}'
             )
         return type(self)(self._internal_representation + other._internal_representation)
+
+
+@dataclass(frozen=True, eq=False)
+class SumOfProducts(AbstractControlValues):
+    """SumOfProducts represents control values in a form of a list or products.
+
+    SumOfProducts allows the creation of control values combinations that
+    can't be factored as the product of independent expressions (e.g. XOR)
+
+
+    Examples:
+        xor = SumOfProducts(((0, 1), (1, 0)))   # xor of two bits.
+        nand =  SumOfProducts(((0, 0), (0, 1), (1, 0))) # nand of two bits
+    """
+
+    _internal_representation: Tuple[Tuple[int, ...], ...]
```

Comment:
Open question, would this be better as FrozenSet in the outer part, to prevent dupes? It would be slightly slower to iterate and slightly larger to store, so there's a good argument not to change it too. If we do, ProductOfSums could be a Frozenset on the inner part too.